src/machine/asm/mod.rs

```diff
@@ -103,6 +124,7 @@ impl Memory<u64> for Box<AsmCoreMachine> {
     }
 
     fn store_bytes(&mut self, addr: u64, value: &[u8]) -> Result<(), Error> {
+        inited_memory(self, addr, value.len() as u64)?;
```

Comment:
Why init memory on a store operation?

Reply:
If we don't do this, the memory will be zeroed the next time `read()`, then the data written will be overwritten.